### PR TITLE
fishnet: 2.5.1 -> 2.7.1

### PR DESCRIPTION
**Pull request by dmytrokyrychuk**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by dmytrokyrychuk** [comment]
Builds and runs with `doCheck = false`. Otherwise fails because `Stockfish/src/${nnueFile}` somehow disappears from the source directory during the build (it's still there after `postPatch`), which causes [`make net`](https://github.com/official-stockfish/Stockfish/blob/0024133b08777b578c1036eb1e4a36343b7fa1bb/src/Makefile#L932-L937) to fail trying to download the file.

**Comment by dmytrokyrychuk** [comment]
I added a line to copy the nnue file again after the build so that the checks that happen next don't fail.

@tu-maurice can please review this?

**Comment by icewind1991** [comment]
Note that whatever caused fishnet to be marked as only supporting x86_64 doesn't seem to be an issue anymore, everything is working fine on on aarch64-linux.

So the platforms limitation can probably be removed.

**Review by icewind1991 — APPROVED** [review]
All works as expected

**Comment by dmytrokyrychuk** [comment]
I found the autoPatchelfHook issue at #113930. Since autoPatchelfHook is no longer used, I will remove the platform restriction.

**Comment by dmytrokyrychuk** [comment]
The package is actually [not building ](https://github.com/NixOS/nixpkgs/pull/263247/checks?check_run_id=18101130236) on darwin, I will limit the platforms to aarch64-linux and x86_64-linux

File: pkgs/servers/fishnet/default.nix [hunk 1]
```diff
@@ -29,19 +29,18 @@ rustPlatform.buildRustPackage rec {
...
+  cargoHash = "sha256-NO3u2ZXSiDQnZ/FFZLOtTnQoGMyN9pSI4sqGIXtjEcI=";
 
   meta = with lib; {
     description = "Distributed Stockfish analysis for lichess.org";
-    homepage = "https://github.com/niklasf/fishnet";
+    homepage = "https://github.com/lichess-org/fishnet";
     license = licenses.gpl3Plus;
     maintainers = with maintainers; [ tu-maurice ];
-    platforms = [ "x86_64-linux" ];
+    platforms = [ "aarch64-linux" "x6_64-linux" ];
```
Comment (tu-maurice) [inline pkgs/servers/fishnet/default.nix:44]: Should be x86_64

**Review by tu-maurice — CHANGES_REQUESTED** [review]

File: pkgs/servers/fishnet/default.nix (hunk 1 above)
Comment (dmytrokyrychuk) [reply pkgs/servers/fishnet/default.nix:44]: Good catch!

**Review by tu-maurice — APPROVED** [review]
Looks good to me. Ran it a few hours without issue.

**Pull request merged**